File: app/src/main/java/com/yao/zhihudaily/net/ZhihuApi.java
package com.yao.zhihudaily.net;

import com.yao.zhihudaily.model.CommentJson;
import com.yao.zhihudaily.model.DailiesJson;
import com.yao.zhihudaily.model.DailyJson;
import com.yao.zhihudaily.model.HotJson;
import com.yao.zhihudaily.model.RecommendsJson;
import com.yao.zhihudaily.model.SectionJson;
import com.yao.zhihudaily.model.SectionsJson;
import com.yao.zhihudaily.model.StartImageJson;
import com.yao.zhihudaily.model.StoryExtra;
import com.yao.zhihudaily.model.ThemeJson;
import com.yao.zhihudaily.model.ThemesJson;

import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Streaming;
import retrofit2.http.Url;

/**
 * Created by Administrator on 2016/9/21.
 */

public interface ZhihuApi {

    @GET("4/start-image/1080*1776")
    Observable<StartImageJson> getStartImage();

    @Streaming//注明为流文件，防止retrofit将大文件读入内存
    @GET
    Observable<ResponseBody> getStartImageFile(@Url String url);//通过@Url覆盖baseurl

    @GET("4/news/latest")
    Observable<DailiesJson> getDailies();

    @GET("4/news/before/{date}")
    Observable<DailiesJson> getDailiesBefore(@Path("date") String date);

    @GET("4/news/{id}")
    Observable<DailyJson> getNews(@Path("id") String id);

    @GET("4/story-extra/{id}")
    Observable<StoryExtra> getStoryExtra(@Path("id") String id);

    @GET("4/story/{id}/short-comments")
    Observable<CommentJson> getShortComments(@Path("id") String id);

    @GET("4/story/{id}/long-comments")
    Observable<CommentJson> getLongComments(@Path("id") String id);

    @GET("4/themes")
    Observable<ThemesJson> getThemes();

    @GET("4/theme/{id}")
    Observable<ThemeJson> getTheme(@Path("id") String id);

    @GET("3/news/hot")
    Observable<HotJson> getHot();

    @GET("3/sections")
    Observable<SectionsJson> getSections();

    @GET("3/section/{id}")
    Observable<SectionJson> getSection(@Path("id") String id);

    @GET("3/section/{id}/before/{timestamp}")
    Observable<SectionJson> getSectionBefore(@Path("id") String id, @Path("timestamp") String timestamp);

    @GET("4/story/{id}/recommenders")
    Observable<RecommendsJson> getRecommends(@Path("id") String id);

    @GET("4/editor/{id}/profile-page/android")
    Observable getEditor(@Path("id") String id);

}

File: app/src/main/java/com/yao/zhihudaily/net/ZhihuApi.kt
package com.yao.zhihudaily.net

import com.yao.zhihudaily.model.*
import io.reactivex.Observable
import okhttp3.ResponseBody
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Streaming
import retrofit2.http.Url

/**
 * @author Yao
 * @date 2016/9/21
 */
interface ZhihuApi {

    @get:GET("4/start-image/1080*1776")
    val startImage: Observable<StartImageJson>

    @get:GET("4/news/latest")
    val dailies: Observable<DailiesJson>

    @get:GET("4/themes")
    val themes: Observable<ThemesJson>

    @get:GET("3/news/hot")
    val hot: Observable<HotJson>

    @get:GET("3/sections")
    val sections: Observable<SectionsJson>

    @Streaming
    @GET
    fun getStartImageFile(@Url url: String): Observable<ResponseBody> //通过@Url覆盖baseurl

    @GET("4/news/before/{date}")
    fun getDailiesBefore(@Path("date") date: String): Observable<DailiesJson>

    @GET("4/news/{id}")
    fun getNews(@Path("id") id: String): Observable<DailyJson>

    @GET("4/story-extra/{id}")
    fun getStoryExtra(@Path("id") id: String): Observable<StoryExtra>

    @GET("4/story/{id}/short-comments")
    fun getShortComments(@Path("id") id: String): Observable<CommentJson>

    @GET("4/story/{id}/long-comments")
    fun getLongComments(@Path("id") id: String): Observable<CommentJson>

    @GET("4/theme/{id}")
    fun getTheme(@Path("id") id: String): Observable<ThemeJson>

    @GET("3/section/{id}")
    fun getSection(@Path("id") id: String): Observable<SectionJson>

    @GET("3/section/{id}/before/{timestamp}")
    fun getSectionBefore(@Path("id") id: String, @Path("timestamp") timestamp: String): Observable<SectionJson>

    @GET("4/story/{id}/recommenders")
    fun getRecommends(@Path("id") id: String): Observable<RecommendsJson>

    @GET("4/editor/{id}/profile-page/android")
    fun getEditor(@Path("id") id: String): Observable<*>

}
